homeassistant/components/xiaomi_miio/fan.py

```diff
@@ -1199,6 +1201,18 @@ def button_pressed(self):
 
         return None
 
+    @property
+    def water_level(self):
+        """Return the water tank levels as percentage."""
+        if self._state:
+            reported_level = self._state_attrs[ATTR_WATER_LEVEL]
+            #  127 without water tank. 120 = 100% water
+            if reported_level == 127:
+                return -1
```

Comment:
Is -1 better than None here?

Reply:
The idea was to differentiate between "tank removed" and "value not reported", but to be honest I wasn't sure if there is any value (no pun intended) in differentiating that. Do you think it should be changed?

Reply:
I don't have a strong opinion but probably it should be consistent with the rest of HA, and I think reporting -1 for unavailable is rare.  

If the water tank is removable, could that be a separate attribute?  I think that would be more helpful to users who might want to trigger an automation for water tank removed/replaced etc.  (For example a notification to refill the water tank could be shown at <5%, then the notification could be hidden once the tank was removed/replaced.).  With the -1 option, the user kind of has to work out what -1 means for themselves

But like I say, no strong opinion my side.

Reply:
Thanks for sharing your view. To be honest, I also don't like special values like `-1` but couldn't think of the better solution. Creating new state attribute or entity seems like a good one.